src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/Streaming/Streaming.tsx

```diff
@@ -0,0 +1,150 @@
+  },
+});
+
+const displayName = 'dashboard.ExpenditurePage.Streaming';
+
+interface Props {
+  sidebarRef: HTMLElement | null;
+  colony: Colony;
+}
+
+const Streaming = ({ colony, sidebarRef }: Props) => {
+  const [, { value: fundingSources }, { setValue }] = useField<
+    StreamingType['fundingSource']
+  >('streaming.fundingSource');
+
+  const newFundingSourceData = useMemo(() => {
+    return {
+      ...{
+        ...newFundingSource,
+        rate: {
+          ...newFundingSource,
+          token: colony?.nativeTokenAddress,
+        },
+      },
+    };
+  }, [colony]);
+
+  const onToggleButtonClick = useCallback(
+    (index) => {
+      setValue(
+        fundingSources?.map((fundingSource, idx) =>
+          index === idx
+            ? { ...fundingSource, isExpanded: !fundingSource.isExpanded }
+            : fundingSource,
+        ),
+      );
+    },
+    [fundingSources, setValue],
+  );
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <FormattedMessage {...MSG.fundingSource} />
+      </div>
+      <FieldArray
+        name="streaming.fundingSource"
+        render={({ push, remove }) => (
+          <>
+            {fundingSources?.map((fundingSource, index) => {
+              const domain = colony?.domains.find(
+                ({ ethDomainId }) => Number(fundingSource.team) === ethDomainId,
+              );
+
+              return (
+                <div
+                  className={styles.singleFundingSource}
+                  key={fundingSource.id}
+                >
+                  <FormSection>
+                    <div className={styles.fundingSourceLabel}>
+                      <CollapseExpandButtons
+                        isExpanded={fundingSource.isExpanded}
+                        onToogleButtonClick={() => onToggleButtonClick(index)}
+                        isLastitem={index === fundingSources?.length - 1}
+                      />
+                      <FormattedMessage
+                        {...MSG.title}
+                        values={{ nr: index + 1, team: domain?.name }}
+                      />
+                      {fundingSources?.length > 1 && (
+                        <Icon
+                          name="trash"
+                          className={styles.deleteIcon}
+                          onClick={() => remove(index)}
+                        />
+                      )}
+                    </div>
+                  </FormSection>
+                  <FundingSource
+                    sidebarRef={sidebarRef}
+                    colony={colony}
+                    index={index}
+                    fundingSource={fundingSource}
+                    isLast={index === fundingSources?.length - 1}
+                  />
+                </div>
+              );
+            })}
+            <Button
+              onClick={() => {
+                push({ ...newFundingSourceData, id: nanoid() });
```

Comment:
There is no need to add id part here, you can add it in the newFundingSourceData useMemo straight away, then there is no need to restructure the object and it's cleaner. Please update.

Reply:
> There is no need to add id part here, you can add it in the newFundingSourceData useMemo straight away, then there is no need to restructure the object and it's cleaner. Please update.

If I move the id to useMemo, it will be the same for all new funding sources. And this causes a unique key error in the console.